### PR TITLE
SCP-2599 - Merge API and NewAPI; now simply API.

### PR DESCRIPTION
**Pull request by silky**
Fixes SCP-2599

- Merge API and NewAPI
- Some minor cleanup

Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
    - [x] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested

### CODE REVIEW

File: plutus-pab/src/Plutus/PAB/Webserver/API.hs [hunk 1]
```diff
@@ -19,37 +17,31 @@ import           Plutus.PAB.Webserver.Types (ContractActivationArgs, ContractIns
...
--- | PAB client API for contracts of type @t@. Examples of @t@ are
---   * Contract executables that reside in the user's file system
+-- | PAB client API for contracts of type @t@. An example of @t@ are
 --   * "Builtin" contracts that run in the same process as the PAB (ie. the PAB is compiled & distributed with these contracts)
-type NewAPI t walletId -- see note [WalletID type in wallet API]
-    = "api" :> "new" :> "contract" :>
-        ("activate" :> ReqBody '[ JSON] (ContractActivationArgs t) :> Post '[JSON] ContractInstanceId -- start a new instance
+type API t walletId -- see note [WalletID type in wallet API]
+    = "api" :> ("healthcheck" :> Get '[JSON] ()) -- Is the server alive?
+    :<|> ("fullreport" :> Get '[JSON] (FullReport t)) -- Details of the contracts: the signatures and their states.
```
Comment (j-mueller) [inline plutus-pab/src/Plutus/PAB/Webserver/API.hs:34]: if I read this correctly it produces

```
GET /api/healthcheck
GET /fullreport
POST /new/contract/activate
...
```

We can delete the "new" bit now and group everything under "api"

```
GET /api/healthcheck
GET /api/fullreport
POST /api/contract/activate
...
```

**Review by j-mueller — CHANGES_REQUESTED** [review]
See inline comment

File: plutus-pab/src/Plutus/PAB/Webserver/API.hs (hunk 1 above)
Comment (silky) [reply plutus-pab/src/Plutus/PAB/Webserver/API.hs:34]: Ah yes of course; nice, thanks I'll fix that.

Comment (silky) [reply plutus-pab/src/Plutus/PAB/Webserver/API.hs:35]: Thanks for showing me that `layout` trick for seeing the paths!

It's now:

```
/
└─ api/
   ├─ contract/
   │  ├─ activate/
   │  │  └─•
   │  ├─ definitions/
   │  │  └─•
   │  ├─ instance/
   │  │  └─ <capture>/
   │  │     ├─ endpoint/
   │  │     │  └─ <capture>/
   │  │     │     └─•
   │  │     ├─ status/
   │  │     │  └─•
   │  │     └─ stop/
   │  │        └─•
   │  └─ instances/
   │     ├─ wallet/
   │     │  └─ <capture>/
   │     │     └─•
   │     └─•
   ├─ fullreport/
   │  └─•
   └─ healthcheck/
      └─•
```

**Review by j-mueller — APPROVED** [review]

**Comment by j-mueller** [comment]
Looks good now, but there may be some changes required to the Purescript clients

@merivale : Can you have a look at this please? (The API layout has changed but the routes are still there)

**Comment by silky** [comment]
@j-mueller Makes sense; I'll try and have a look as well.

**Comment by silky** [comment]
@j-mueller @merivale - I fixed up the `plutus-pab-client` bits; so it should be good to go now :)

**Pull request merged**